### ISSUE TITLE
"sign in with Email" crashes with NEW email address and on iOS only, but no log error message to help?

### ISSUE THREAD

**Issue by fpisidoro** [koji-1009/flutter_auth_ui#6]
This is an excellent package that is working well, but I am having one major problem which is difficult for me to solve because there is no error message, just a crash. Therefore I don't know which component has the problem, such as if it is a problem with the firebase_auth package.

Can anyone confirm on iOS if you are able to successfully sign in with email with a NEW account (i.e. create an account)?

Otherwise, the FirebaseUI works fine:
- iOS "sign in with Google", both device and simulator: Successful
- iOS "sign in with Apple", both device and simulator: Successful
- iOS "sign in with Email" using already registered email address, both device and simulator: Successful
- Android "sign in with Email" using already registered email address, both device and simulator: Successful
- Android "sign in with Email" using _new_ email address, both device and simulator: Successful
- iOS "sign in with Email" using _new_ email address, on **device**: CRASH. The app justs quits immediately.
- iOS "sign in with Email" using _new_ email address, on **simulator**: CRASH. The app justs quits immediately. There is nothing in the console log at all.

The log only says:
```
Lost connection to device.
Exited (sigterm)
```

That's it. Nothing else.
Note that this only happens on iOS, yet using a known email works fine, and the other iOS sign in methods (Google, Apple) work fine.

If someone can confirm if they are able to log in with a NEW unregistered email address, then that would mean it is my fault with a configuration somewhere. If someone can confirm a crash when entering a new email address, then how do I track down what it causing it if there is no error message?

**Comment by koji-1009** [comment]
Hi!
The crash also happened in my environment.

I got the crash log. 
ref <https://firebase.google.com/docs/analytics/debugview#ios>

```
2020-08-20 02:07:07.604809+0900 Runner[46129:876598] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** +[NSURLComponents initWithString:]: nil URLString parameter'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23e3de6e __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff512539b2 objc_exception_throw + 48
	2   Foundation                          0x00007fff259988ce -[__NSConcreteURLComponents URL] + 0
	3   Foundation                          0x00007fff2599a2ac +[NSURLComponents componentsWithString:] + 33
	4   Runner                              0x00000001081b10dc -[FUIEmailAuth generateURLParametersAndLocalCache:linkingProvider:] + 236
	5   Runner                              0x00000001081b3675 -[FUIEmailEntryViewController sendSignInLinkToEmail:] + 309
	6   Runner                              0x00000001081b308e __38-[FUIEmailEntryViewController onNext:]_block_invoke + 1822
	7   Runner                              0x000000010811cfbd __49-[FIRAuth fetchSignInMethodsForEmail:completion:]_block_invoke_3 + 93
	8   libdispatch.dylib                   0x000000010b6a4f11 _dispatch_call_block_and_release + 12
	9   libdispatch.dylib                   0x000000010b6a5e8e _dispatch_client_callout + 8
	10  libdispatch.dylib                   0x000000010b6b3d97 _dispatch_main_queue_callback_4CF + 1149
	11  CoreFoundation                      0x00007fff23da1869 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	12  CoreFoundation                      0x00007fff23d9c3b9 __CFRunLoopRun + 2041
	13  CoreFoundation                      0x00007fff23d9b8a4 CFRunLoopRunSpecific + 404
	14  GraphicsServices                    0x00007fff38c39bbe GSEventRunModal + 139
	15  UIKitCore                           0x00007fff49325968 UIApplicationMain + 1605
	16  Runner                              0x000000010811869b main + 75
	17  libdyld.dylib                       0x00007fff520ce1fd start + 1
	18  ???                                 0x0000000000000002 0x0 + 2
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

Perhaps, there is a problem with the following code.
<https://github.com/koji-1009/flutter_auth_ui/blob/0.2.2/ios/Classes/SwiftFlutterAuthUiPlugin.swift#L60>

I will investigate that.
Please give me some time.

**Comment by koji-1009** [comment]
https://github.com/koji-1009/flutter_auth_ui/blob/master/lib/flutter_auth_ui.dart#L12

```
  /// If you want iOS support, set following params your `Info.plist`.
  /// (see <https://firebase.google.com/docs/auth/ios/firebaseui#email_link_authentication>)
  ///   - FirebaseAuthUiEmailHandleURL : String, set `ActionCodeSettings.url`
  ///   - FirebaseAuthUiEmailAndroidPackageName : String, set `actionCodeSettings.setAndroidPackageName` - packageName
  ///   - FirebaseAuthUiEmailAndroidMinimumVersion : Int, set `actionCodeSettings.setAndroidPackageName` - minimumVersion
```

Could you check your `Info.plist`?

This only affects iOS, so I designed it to be set in Info.plist.
However, this mechanism can be confusing. Sorry.

**Comment by fpisidoro** [comment]
Yes, thank you. I recently did find these settings for info.plist and have been experimenting, with partial success. I am having trouble with the firebase dynamic link part of the configuration apparently required for email link authentication.

1. Am I correct that these three parameters to Info.plist are for email **link** authentication only? If I add these three parameters, then the firebase-UI does _not_ allow me to do email+**password** authentication for a new email address (a dialog box error message tells me to turn _on_ email-link-auth in firebase console). Is there an Info.plist parameter that will satisfy email-**password** auth?

2. I am confused as to what I actually put in the Info.plist. for **FirebaseAuthUiEmailHandleURL** I tried to put a dynamic link created in the Firebase console, but I have not actually been able to successfully complete a sign up process using a new email. For **FirebaseAuthUiEmailAndroidPackageName** I put the android package name listed in the Firebase console (and not the iOS bundle ID. For **FirebaseAuthUiEmailAndroidMinimumVersion** I just put "12" because that's what some examples online use. Am I doing this correctly?

I will continue to experiment with these settings. Thank you.
